shared-libs/transitions/src/transitions/generate_patient_id_on_people.js

```diff
@@ -3,14 +3,23 @@ const generateShortcodeOnContacts = require('./generate_shortcode_on_contacts');
 
 // As of 3.8.0, generates a shortcode for every configured contact type.
 module.exports = {
+  name: 'generate_patient_id_on_people',
+  asynchronousOnly: true,
+  deprecated: true,
+  deprecatedIn: '3.8.x',
   init: () => {
-    logger.warn(
-      '"generate_patient_id_on_people" transition is deprecated. ' +
-      'Please use "generate_shortcode_on_contacts" transition instead.'
-    );
+    const self = module.exports;
+
+    if (self.deprecated) {
+      logger.warn(self.getDeprecationMessage());
+    }
   },
+  getDeprecationMessage: () => {
```

Comment:
If the direction is to have a "standard" deprecation message and add additional information tailored to each transition, I think we should add the "constructor" of this deprecation message in a utils library (for example, here https://github.com/medic/cht-core/blob/master/shared-libs/transitions/src/transitions/utils.js) , to avoid all this code duplication.

Reply:
Done. Moved to utils and unit tested.